src/main/java/bio/terra/service/filedata/google/firestore/FireStoreUtils.java

```diff
@@ -299,6 +306,7 @@ static boolean shouldRetry(Throwable throwable, boolean isBatch) {
         || throwable instanceof InternalException
         || throwable instanceof StatusRuntimeException
         || throwable instanceof GoogleResourceException
+        || throwable instanceof StorageException
```

Comment:
It's a little odd that there are exceptions in this list that aren't in the outer `catch` clause, but I assume that's because they only occur as causes, not top level exceptions?

Reply:
I think you're exactly right - we want to loop through both the top level exceptions and causes to check for retryable exceptions.